Draft: feat: Update default font and update some colors

rockerBOO opened the pull request:
- Rehashed the primary button on the homepage
- Updated font to use Inter
- Added some padding to the sections.

On colors, balanced the dark mode for links to be a little less colorful. Makes it easier to read when the links are not too jarring. 

On fonts, changed from Lato to Inter mostly because of the underline spacing is too close. I used it for both titles and longer text areas like the news and homepage but I also have a version with a serif, IBM Plex Serif that looks pretty good.

Header sizes I reduced from 2rem for h1 to 1.6em for h1. I feel I might need to look at these some more but this is how they are now.

Updated main button to be less dull and chose colors that work on dark and light with higher saturation.

All colors should still exceed AAA color contrast.

![Screenshot 2022-10-29 at 17-48-40 Home - Neovim](https://user-images.githubusercontent.com/15027/198854073-0b67660f-4a1a-458f-aa92-6ad6bd2886c3.png)
![Screenshot 2022-10-29 at 17-48-33 Home - Neovim](https://user-images.githubusercontent.com/15027/198854076-03e8468d-f894-4f4a-ad85-4ba323cc6261.png)

justinmk commented:
The default font size is too small. Can we increase it, the size used for the docs seems better (I manually increased it there) : https://neovim.io/doc/user/api.html

rockerBOO commented:
Upping size on larger screen/viewports

```css
@media (min-width: 50em) {
	body {
		font-size: 1.1rem;
	}
}
```

`1.1rem`
![Screenshot 2022-10-30 at 11-35-58 Home - Neovim](https://user-images.githubusercontent.com/15027/198887480-b918240a-8158-40a0-9ade-505f11d794f2.png)

rockerBOO commented:
Updated links, 
- link underline color on dark
- link underline when hovered, focused

Masthead
- Removed border from masthead. 
- Added variable height padding based on viewport to masthead.

![Screenshot 2022-10-30 at 12-58-51 Home - Neovim](https://user-images.githubusercontent.com/15027/198891316-89eda552-d606-4155-a5e5-7f6a113f84a3.png)
![Screenshot 2022-10-30 at 12-58-24 Home - Neovim](https://user-images.githubusercontent.com/15027/198891317-7e554b4f-8f53-48ce-94da-970a877735b6.png)

rockerBOO commented:
All ready to go. Would love to have this up before neovimconf on the 9th. Can make any changes, and we can iterate into the future. Thanks.

justinmk commented on css/main.css:
```diff
@@ -1,32 +1,56 @@
+  --serif-fonts: "Inter", var(--bs-font-serif);
   /* Sans-serif for smoother fonts. Useful for titles and shorter chunks */
-  --sans-serif-fonts: Lato, sans-serif;
+  --sans-serif-fonts: "Inter", var(--bs-font-sans-serif);
```
The old font looked a bit cheap to me, will be great if this new one feels better :)
